lib/stat_tracker.rb

```diff
@@ -54,57 +54,37 @@ def lowest_scoring_home_team
     end
   end
 
-  def highest_scoring_visitor 
-    #make a hash of of total away games by team id
-    away_games_by_team_id = @games.each_with_object(Hash.new(0.0)) do |game, hash|
-      hash[game.away_team_id] += 1
-    end
-    #make a hash of total away goals by team id
-    away_goals_by_team_id = @games.each_with_object(Hash.new(0.0)) do |game, hash|
-      hash[game.away_team_id] += game.away_goals
+  def total_away_goals 
+    total_goals = @game_teams.each_with_object({}) do |game, hash|
+      hash[game.team_id] = hash[game.team_id] || [0, 0]
+      hash[game.team_id] = [game.goals + hash[game.team_id][0], hash[game.team_id][1] + 1] if game.hoa == "away"
     end
-    
-    #average them together, very similar to the average we did with #average_goals_by_season
-    average_away_goals_by_team_id = Hash.new(0.0)
-    away_goals_by_team_id.each do |key, value|
-      average_away_goals_by_team_id[key] = (value/ away_games_by_team_id[key]).round(2)
+  end
+
+  def highest_scoring_visitor
+    avg_goals = total_away_goals.transform_values do |value|
+      (value[0] / value[1].to_f).round(4)
     end
-    average_away_goals_by_team_id
-    #find the lowest scoring team key/value pair and return only the key because
-    #this produces at array
-    highest_average_scoring_team_id = average_away_goals_by_team_id.max_by {|k, v| v}.first
+    highest_avg_goals = avg_goals.values.max
     
-    #iterate through @teams data and find which team has the id that matched to the
-    #previous variable
-    highest_scoring_team = @teams.find do |team|
-      team.team_name if team.team_id == highest_average_scoring_team_id
+    highest_team_id = avg_goals.key(highest_avg_goals)
+
+    @teams.each do |team| 
+      return team.team_name if team.team_id == highest_team_id
     end
-    team = highest_scoring_team.team_name
   end
 
-    #I reused all the code from above, but only changes the names of the last variables
-    #to say "lowest"
-  def lowest_scoring_visitor 
-    away_games_by_team_id = @games.each_with_object(Hash.new(0.0)) do |game, hash|
-      hash[game.away_team_id] += 1
+  def lowest_scoring_visitor
+    avg_goals = total_away_goals.transform_values do |value|
```

Comment:
Seems like #avg_goals can be pulled out like you mentioned. I have the same method helper so that can be shared across our four methods. Other than that the code is great.